### PR TITLE
nuScenes lidar segmentation challenge

### PR DESCRIPTION
**Pull request by whyekit-aptiv**
### This PR will:
- Set out the rules and details for the nuScenes-lidarseg challenge

### CODE REVIEW

File: python-sdk/nuscenes/eval/lidarseg/README.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+We then define boundary IOU as TP / (TP + FP + FN) but taken only on the boundary "pixels".
+This is inspired by the work done by [Milioto et al.](http://www.ipb.uni-bonn.de/wp-content/papercite-data/pdf/milioto2019iros.pdf)
+Note that the boundary IOU is class-agnostic, i.e. it is **not** evaluated per class.
+
+### nuScenes-lidarseg score
+* **nuScenes-lidarseg score (NLS)**:
+We consolidate the above metrics by taking the average of mIOU and bIOU (i.e. equal weighting is given to both).
+
+### Configuration
+The default evaluation metrics configurations can be found in `nuscenes/eval/detection/configs/detection_cvpr_2019.json`. ## TODO
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/lidarseg/README.md:167]: Remove unless needed

File: python-sdk/nuscenes/eval/lidarseg/README.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+Note that this restriction applies only at test time.
+At training time any sensor input may be used.
+
+* *Map data:*
+By `map data` we mean using the *semantic* map provided in nuScenes. 
+
+* *Meta data:*
+Other meta data included in the dataset may be used without restrictions.
+E.g. calibration parameters, ego poses, `location`, `timestamp`, `num_lidar_pts`, `num_radar_pts`, `translation`, `rotation` and `size`.
+Note that `instance`, `sample_annotation` and `scene` description are not provided for the test set.
```
Comment (holger-nutonomy) [inline python-sdk/nuscenes/eval/lidarseg/README.md:200]: Mention that nuScenes box annotations may be used during training.
Adapt for nuScenes-lidarseg. Add the bin files, besides `sample_annotation`.

File: python-sdk/nuscenes/eval/lidarseg/README.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+  ```
+  "meta": {
+      "use_camera":   <bool>          -- Whether this submission uses camera data as an input.
+      "use_lidar":    <bool>          -- Whether this submission uses lidar data as an input.
+      "use_radar":    <bool>          -- Whether this submission uses radar data as an input.
+      "use_map":      <bool>          -- Whether this submission uses map data as an input.
+      "use_external": <bool>          -- Whether this submission uses external data as an input.
+  },
+  "mapping": {
+      class_idx <int>: class_name <str> -- Maps each class index to a class name.
```
[the remark this answers is not in the document]
Comment (whyekit-aptiv) [reply python-sdk/nuscenes/eval/lidarseg/README.md:84]: As discussed, we will standardize the mapping (listed in the section on [Classes](https://github.com/nutonomy/nuscenes-devkit/pull/470/files#diff-15eee3e2a396cc5579d63b9e290babef48f5a1b15e8a65942216b3e7e0b41938R108) below)

[the remark this answers is not in the document]
Comment (holger-nutonomy) [reply python-sdk/nuscenes/eval/lidarseg/README.md:84]: Can we provide the actual json example for reference?

**Review by holger-nutonomy — APPROVED** [review]

File: python-sdk/nuscenes/eval/lidarseg/README.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+  [1, 5, 4, 1, ...]
+  ```
+  Each `lidar_sample_data_token` from the current evaluation set must be included in the `v1.0-test` folder.
+  
+For the train and val sets, the evaluation can be performed by the user on their local machine.
+For the test set, the user needs to zip the results folder and submit it to the official evaluation server.
+
+Note that the lidar segmentation classes may differ from the general nuScenes classes, as detailed below.
+
+## Classes
```
[the remark this answers is not in the document]
Comment (oscar-nutonomy) [reply python-sdk/nuscenes/eval/lidarseg/README.md:96]: looks good to me.

**Review by oscar-nutonomy — APPROVED** [review]

**Pull request merged**